### PR TITLE
Add pod security policy to SDS task.

### PR DESCRIPTION
**Pull request by yangminzhu**
@rcaballeromx Could you help review the doc? I will send another PR for `master` later.

/cc @wenchenglu @wattli @quanjielin 

Signed-off-by: Yangmin Zhu <ymzhu@google.com>

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -494,6 +494,7 @@ uncomment
 uncommented
 Undeploy
 unencrypted
+unix
```
Comment (rcaballeromx) [inline .spelling]: Please use proper capitalization in the text, Unix, and the spell checker should not complain
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 2]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+1. Enable pod security policy in the cluster
+
+    Different cloud platforms have different approach to enable the pod security policy, please refer to your
+    cloud platform for detailed instructions. For GCP, please follow
+    [Enabling pod security policy controller](https://cloud.google.com/kubernetes-engine/docs/how-to/pod-security-policies#enabling_podsecuritypolicy_controller).
+
+    {{< warning >}}
+    Please make sure all needed permission has been granted by the pod security policy before enabling the pod
+    security policy.
+    Once pod security policy is enabled, a pod will fail to start if it requires any permissions not in the
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
    Once the policy is enabled, pods won't start if  they require any permissions not granted.
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 3]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+    pod security policy.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 4]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+    {{< /warning >}}
+
+1. Verify the citadel agent works with the pod security policy
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. Run the following command to restart the Citadel agents:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 5]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    Run the following command to restart the citadel agents.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 6]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    {{< text bash >}}
+    $ kubectl delete pod -l 'app=nodeagent' -n istio-system
+    pod "istio-nodeagent-dplx2" deleted
+    pod "istio-nodeagent-jrbmx" deleted
+    pod "istio-nodeagent-rz878" deleted
+    {{< /text >}}
+
+    Wait a few seconds and run the following command to confirm the citadel agent started successfully.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. To verify that the Citadel agents work with the enabled pod security policy, wait a few seconds
   and run the following command to confirm the citadel agent started successfully:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 7]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    {{< text bash >}}
+    $ kubectl get pod -l 'app=nodeagent' -n istio-system
+    NAME                    READY   STATUS    RESTARTS   AGE
+    istio-nodeagent-p4p7g   1/1     Running   0          4s
+    istio-nodeagent-qdwj6   1/1     Running   0          5s
+    istio-nodeagent-zsk2b   1/1     Running   0          14s
+    {{< /text >}}
+
+1. Verify the normal pod works with the pod security policy
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. Run the following command to start a normal pod.
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 8]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    Run the following command to start a normal pod.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 9]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+        spec:
+          containers:
+          - name: normal
+            image: pstauffer/curl
+            command: ["/bin/sleep", "3650d"]
+            imagePullPolicy: IfNotPresent
+    EOF
+    {{< /text >}}
+
+    Wait a few seconds and run the following command to confirm the normal pod started successfully.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. To verify that the normal pod works with the pod security policy
   enabled, wait a few seconds and run the following command to
   confirm the normal pod started successfully.
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 10]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+              mountPath: /var/run/sds
+          volumes:
+          - name: sds-uds
+            hostPath:
+              path: /var/run/sds
+              type: ""
+    EOF
+    {{< /text >}}
+
+    Run the following command to confirm the malicious pod failed to start due to the pod security policy.
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. To verify that the Unix domain socket is protected, run the following command to confirm the malicious pod failed to start due to the pod security policy:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 11]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    Wait a few seconds and run the following command to confirm the normal pod started successfully.
+
+    {{< text bash >}}
+    $ kubectl get pod -l 'app=normal'
+    NAME                      READY   STATUS    RESTARTS   AGE
+    normal-64c6956774-ptpfh   2/2     Running   0          8s
+    {{< /text >}}
+
+1. Verify the unix domain socket is protected by the pod security policy
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 12]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    Run the following command to start a malicious pod which tries to mount the unix domain socket with
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. Start a malicious pod that tries to mount the Unix domain socket using a
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 13]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+    $ kubectl describe rs -l 'app=malicious' | grep Failed
+    Pods Status:    0 Running / 0 Waiting / 0 Succeeded / 0 Failed
+      ReplicaFailure   True    FailedCreate
+      Warning  FailedCreate  4s (x13 over 24s)  replicaset-controller  Error creating: pods "malicious-7dcfb8d648-" is forbidden: unable to validate against any pod security policy: [spec.containers[0].volumeMounts[0].readOnly: Invalid value: false: must be read-only]
+    {{< /text >}}
+
 ## Cleanup
 
-Clean up test services and Istio control plane:
+1. Clean up test services and Istio control plane:
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. Clean up the test services and the Istio control plane:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 14]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
-$ kubectl delete ns bar
-$ kubectl delete -f istio-auth-sds.yaml
-{{< /text >}}
+    {{< text bash >}}
+    $ kubectl delete ns foo
+    $ kubectl delete ns bar
+    $ kubectl delete -f istio-auth-sds.yaml
+    {{< /text >}}
+
+1. Disable the pod security policy in the cluster:
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
1. Disable the pod security policy in the cluster using the documentation of your platform. If you are using GKE,
   [disable the pod security policy controller](https://cloud.google.com/kubernetes-engine/docs/how-to/pod-security-policies#disabling_podsecuritypolicy_controller).
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 15]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    Different cloud platform has different approach to disable the pod security policy, please refer to your
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 16]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+    cloud platform for detail instructions. For GCP, please follow
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 17]
```diff
@@ -93,15 +93,276 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+    [Disabling pod security policy controller](https://cloud.google.com/kubernetes-engine/docs/how-to/pod-security-policies#disabling_podsecuritypolicy_controller).
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 18]
```diff
@@ -93,15 +93,262 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
...
+
+    {{< text bash >}}
+    $ kubectl delete pod -l 'app=nodeagent' -n istio-system
+    pod "istio-nodeagent-dplx2" deleted
+    pod "istio-nodeagent-jrbmx" deleted
+    pod "istio-nodeagent-rz878" deleted
+    {{< /text >}}
+
+1. To verify that the Citadel agents work with the enabled pod security policy, wait a few seconds
+   and run the following command to confirm the citadel agent started successfully:
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
   and run the following command to confirm the agents started successfully:
```

File: content/docs/tasks/security/auth-sds/index.md [hunk 19]
```diff
@@ -93,15 +93,262 @@ $ kubectl exec -it $(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..me
 
 As you can see there is no secret file mounted at `/etc/certs` folder.
 
+## Increasing security with pod security policies
+
+The Istio Secret Discovery Service (SDS) uses the Citadel agent to distribute the certificate to the
+Envoy sidecar via Unix domain socket. All pods running in the same Kubernetes node share the Citadel
```
Comment (rcaballeromx) [inline content/docs/tasks/security/auth-sds/index.md]:
```suggestion
Envoy sidecar via a Unix domain socket. All pods running in the same Kubernetes node share the Citadel
```

**Comment by yangminzhu** [comment]
@rcaballeromx Comments addressed, thanks for the fast review!

**Pull request merged**